src/azure-cli/azure/cli/command_modules/appconfig/_featuremodels.py

```diff
@@ -129,112 +183,570 @@ def __repr__(self):
         }
         return json.dumps(featurefilter, indent=2, ensure_ascii=False)
 
-# Feature Flag Helper Functions #
 
+class FeatureVariant:
+    """
+    Feature variants class.
 
-def custom_serialize_conditions(conditions_dict):
-    '''
-        Helper Function to serialize Conditions
+    :ivar str name:
+        Name of the variant
+    :ivar str configuration_Value:
+        The configuration value of the variant
+    :ivar str Status_Overide:
+        Overrides the enabled state of the feature if the given variant is assigned.
+        Does not override the state if value is None
+    """
+
+    def __init__(self, name, configuration_value=None, status_override=None):
+        self.name = name
+        if configuration_value is not None:
+            self.configuration_value = configuration_value
+        if status_override is not None:
+            self.status_override = status_override
+
+    @classmethod
+    def convert_from_json(cls, json_string):
+        """
+        Convert JSON string to FeatureVariant object
+
+        Args:
+            json_string - JSON string
+
+        Return:
+            FeatureVariant object
+        """
+        variant = json.loads(json_string)
+        name = variant.get(FeatureFlagConstants.NAME, None)
+        if not name:
+            raise ValidationError(
+                "Feature variant must contain required '%s' attribute: \n%s"
+                % (FeatureFlagConstants.NAME, json.dumps(variant, indent=2, ensure_ascii=False))
+            )
+
+        configuration_value = variant.get(
+            FeatureFlagConstants.VARIANT_CONFIGURATION_VALUE, None
+        )
+        status_override = variant.get(
+            FeatureFlagConstants.VARIANT_STATUS_OVERRIDE, None
+        )
+        return cls(
+            name=name,
+            configuration_value=configuration_value,
+            status_override=status_override,
+        )
+
+    def __repr__(self):
+        featurevariant = {
+            FeatureFlagConstants.NAME: self.name,
+        }
+
+        if hasattr(self, FeatureFlagConstants.VARIANT_CONFIGURATION_VALUE):
+            featurevariant[FeatureFlagConstants.VARIANT_CONFIGURATION_VALUE] = self.configuration_value
+
+        if hasattr(self, FeatureFlagConstants.VARIANT_STATUS_OVERRIDE):
+            featurevariant[FeatureFlagConstants.VARIANT_STATUS_OVERRIDE] = self.status_override
+
+        return json.dumps(featurevariant, indent=2, ensure_ascii=False)
+
+
+class FeatureUserAllocation:
+    """
+    Feature user allocation class.
+
+    :ivar str variant:
+        The name of the variant to use if the user allocation matches the current user.
+    :ivar list users:
+        Collection of users where if any match the current user, the variant specified in the user allocation is used.
+    """
+
+    def __init__(self, variant, users):
+        self.variant = variant
+        self.users = users
+
+    @classmethod
+    def convert_from_json(cls, json_string):
```

Comment:
Correct me if I am wrong but it seems  in places where we are using `convert_from_json`, we are always converting some dictionary to json string first, and then we parse the string again in this method. Can these methods directly accept a dict?

Reply:
Updated to always accept dict